ambry-frontend/src/main/java/com/github/ambry/frontend/PostBlobHandler.java

```diff
@@ -322,6 +322,16 @@ private void setSignedIdMetadataAndBlobSize(BlobProperties blobProperties) throw
             RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.SESSION, true));
         metadata.put(EXPIRATION_TIME_MS_KEY,
             Long.toString(Utils.addSecondsToEpochTime(time.milliseconds(), blobProperties.getTimeToLiveInSeconds())));
+        if (blobProperties.getReservedMetadataBlobId() != null) {
+          metadata.put(RestUtils.Headers.RESERVED_METADATA_ID, blobProperties.getReservedMetadataBlobId());
+        } else {
+          ReservedMetadataIdMetrics.getReservedMetadataIdMetrics(
+              frontendMetrics.getMetricRegistry()).noReservedMetadataFoundForChunkedUploadResponseCount.inc();
+          if (router.getRouterConfig().routerReservedMetadataEnabled) {
+            throw new RestServiceException("No reserved metadata id present to set in chunked upload response",
```

Comment:
If I understand the code logic correctly, setSignedIdMetadataAndBlobSize is called in the PutBlobCallback.
I suppose we have checked the metadata blob exists beforehand already if routerReservedMetadataEnabled is enabled, right? Is that possible we hit getReservedMetadataBlobId == null ?

Reply:
Most likely this will never happen. But I want to cover all cases and make sure we have evidence that everything is working properly for a few weeks before I enabled this feature.